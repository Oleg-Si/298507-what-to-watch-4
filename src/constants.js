export const DEFAULT_GENRE = `All genres`;
export const DEFAULT_TAB = `Overview`;
export const FILM_COUNT = 40;
export const RENDERED_FILMS_COUNT = 8;
export const COUNT_MORE_FILMS = 8;
export const COUNT_MORE_LIKE_THIS_FILMS = 4;

export const Tabs = [
  `Overview`,
  `Details`,
  `Reviews`
];

export const RatingKey = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};

export const Screens = {
  MAIN: `Main`,
  FILM_PAGE: `FilmPage`
};
