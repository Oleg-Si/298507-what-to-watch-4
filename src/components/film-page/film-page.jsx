import React from 'react';
import PropTypes from 'prop-types';
import TabList from '../tab-list/tab-list.jsx';
import {Tabs, COUNT_MORE_LIKE_THIS_FILMS} from '../../constants';
import {connect} from 'react-redux';
import appActionCreator from './../../redux/app/action-creator';
import FilmPageOverview from './../film-page-overview/film-page-overview.jsx';
import FilmPageDetalis from '../film-page-details/film-page-details.jsx';
import FilmPageReviews from '../film-page-reviews/film-page-reviews.jsx';
import FilmList from './../film-list/film-list.jsx';
import {getActiveTab} from '../../redux/app/selectors.js';
import {getFilms, getFilmComments} from './../../redux/data/selectors';
import AppHeader from '../app-header/app-header.jsx';
import {getAuthorizationStatus, getUserAvatar} from './../../redux/user/selectors';
import {Screens, AuthorizationStatus} from './../../constants';

const FilmPage = (props) => {
  const filmInfo = props.activeFilm;
  const {
    activeTab,
    onTabClick,
    onFilmCardTitleClick,
    films,
    authorizationStatus,
    onSignInClick,
    userAvatar,
    onAddReview,
    filmReviews
  } = props;

  const getFilmInfo = () => {
    switch (activeTab) {
      case Tabs.OVERVIEW:
        return <FilmPageOverview
          filmInfo={filmInfo}
        />;

      case Tabs.DETAILS:
        return <FilmPageDetalis
          filmInfo={filmInfo}
        />;

      case Tabs.REVIEWS:
        return <FilmPageReviews
          filmReviews={filmReviews}
        />;
    }

    return null;
  };

  const getFilmsByCurrentGenre = () => {
    const filteredFilms = films.filter((el) => el.genre === filmInfo.genre);
    const currentFilmIndex = filteredFilms.findIndex((el) => el.id === filmInfo.id);

    // Удаляем текущий фильм из похожих
    filteredFilms.splice(currentFilmIndex, 1);

    return filteredFilms;
  };

  const getButtonAddReview = () => {
    if (authorizationStatus === AuthorizationStatus.AUTH) {
      return <a href="add-review.html" className="btn movie-card__button" onClick={(evt) => {
        evt.preventDefault();
        onAddReview();
      }}>Add review</a>;
    }

    return null;
  };

  return (
    <React.Fragment>
      <section className="movie-card movie-card--full" style={{backgroundColor: filmInfo.bgColor}}>
        <div className="movie-card__hero">
          <div className="movie-card__bg">
            <img src={filmInfo.bgImg} alt={filmInfo.title} />
          </div>

          <h1 className="visually-hidden">WTW</h1>

          <AppHeader
            authorizationStatus={authorizationStatus}
            userAvatar={userAvatar}
            onSignIn={onSignInClick}
          />

          <div className="movie-card__wrap">
            <div className="movie-card__desc">
              <h2 className="movie-card__title">{filmInfo.title}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{filmInfo.genre}</span>
                <span className="movie-card__year">{filmInfo.releaseDate}</span>
              </p>

              <div className="movie-card__buttons">
                <button className="btn btn--play movie-card__button" type="button">
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use xlinkHref="#play-s"></use>
                    <symbol id="play-s" viewBox="0 0 19 19">
                      <path fillRule="evenodd" clipRule="evenodd" d="M0 0L19 9.5L0 19V0Z" fill="#EEE5B5" />
                    </symbol>
                  </svg>
                  <span>Play</span>
                </button>
                <button className="btn btn--list movie-card__button" type="button">
                  <svg viewBox="0 0 19 20" width="19" height="20">
                    <use xlinkHref="#add"></use>
                  </svg>
                  <span>My list</span>
                </button>

                {getButtonAddReview()}

              </div>
            </div>
          </div>
        </div>

        <div className="movie-card__wrap movie-card__translate-top">
          <div className="movie-card__info">
            <div className="movie-card__poster movie-card__poster--big">
              <img src={filmInfo.poster} alt={filmInfo.title} width="218" height="327" />
            </div>

            <div className="movie-card__desc">
              <nav className="movie-nav movie-card__nav">

                <TabList
                  tabs={Object.values(Tabs)}
                  activeTab={activeTab}
                  onTabClick={onTabClick}
                />

              </nav>

              {getFilmInfo()}

            </div>
          </div>
        </div>
      </section>

      <div className="page-content">
        <section className="catalog catalog--like-this">
          <h2 className="catalog__title">More like this</h2>

          <FilmList
            films={getFilmsByCurrentGenre()}
            filmsCount={COUNT_MORE_LIKE_THIS_FILMS}
            onFilmCardTitleClick={onFilmCardTitleClick}
          />

        </section>

        <footer className="page-footer">
          <div className="logo">
            <a href="main.html" className="logo__link logo__link--light">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </a>
          </div>

          <div className="copyright">
            <p>© 2020 What to watch Ltd.</p>
          </div>
        </footer>
      </div>
    </React.Fragment>
  );
};

FilmPage.propTypes = {
  activeFilm: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    releaseDate: PropTypes.number.isRequired,
    ratingCount: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    bgImg: PropTypes.string.isRequired,
    bgColor: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(
        PropTypes.string
    ).isRequired
  }).isRequired,
  activeTab: PropTypes.string.isRequired,
  onTabClick: PropTypes.func.isRequired,
  films: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        img: PropTypes.string.isRequired
      })
  ).isRequired,
  onFilmCardTitleClick: PropTypes.func.isRequired,
  onAddReview: PropTypes.func.isRequired,
  onSignInClick: PropTypes.func.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
  userAvatar: PropTypes.string,
  filmReviews: PropTypes.array
};

const mapStateToProps = (state) => ({
  activeTab: getActiveTab(state),
  films: getFilms(state),
  authorizationStatus: getAuthorizationStatus(state),
  userAvatar: getUserAvatar(state),
  filmReviews: getFilmComments(state)
});

const mapDispatchToProps = (dispatch) => ({
  onTabClick(newTab) {
    dispatch(appActionCreator.filmsPageTabChange(newTab));
  },

  onSignInClick() {
    dispatch(appActionCreator.changeScreen(Screens.SIGN_IN));
  },

  onAddReview() {
    dispatch(appActionCreator.changeScreen(Screens.ADD_REVIEW));
  }
});

export {FilmPage};
export default connect(mapStateToProps, mapDispatchToProps)(FilmPage);
