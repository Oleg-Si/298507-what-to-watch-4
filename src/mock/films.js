import {getRandomInt, getRandomArrayItems} from './../utils';
import {FILM_COUNT} from '../constants';

const getFilms = (count) => {
  const allFilms = new Array(count).fill(``);

  allFilms.forEach((el, i, arr) => {
    const element = getMockFilm();
    element.id = i;
    arr[i] = element;
  });

  return allFilms;
};

const getMockFilm = () => ({
  id: 0,
  title: [
    `Fantastic Beasts: The Crimes of Grindelwald`,
    `Bohemian Rhapsody`,
    `Macbeth`,
    `Aviator`,
    `We need to talk about Kevin`,
    `What We Do in the Shadows`,
    `Revenant`,
    `Johnny English`,
    `Shutter Island`,
    `Pulp Fiction`,
    `No Country for Old Men`,
    `Snatch`
  ][Math.floor(Math.random() * 12)],
  img: [
    `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    `img/bohemian-rhapsody.jpg`,
    `img/macbeth.jpg`,
    `img/aviator.jpg`,
    `img/we-need-to-talk-about-kevin.jpg`,
    `img/what-we-do-in-the-shadows.jpg`,
    `img/revenant.jpg`,
    `img/johnny-english.jpg`,
    `img/shutter-island.jpg`,
    `img/pulp-fiction.jpg`,
    `img/no-country-for-old-men.jpg`,
    `img/snatch.jpg`
  ][Math.floor(Math.random() * 12)],
  poster: ``,
  genre: [
    `Dramas`,
    `Horror`,
    `Comedies`,
    `Romance`,
    `Crime`,
    `Sci-Fi`,
    `Documentary`,
    `Thrillers`,
    `Kids & Family`
  ][getRandomInt(0, 6)],
  src: getRandomArrayItems([`https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4`, `https://interactive-examples.mdn.mozilla.net/media/examples/flower.mp4`, `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`, `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`, `https://archive.org/download/BigBuckBunny_124/Content/big_buck_bunny_720p_surround.mp4`], 1),
  rating: `${getRandomInt(0, 9)}.${getRandomInt(0, 9)}`,
  releaseDate: getRandomInt(1262293200000, 1552915226386),
  runTime: getRandomInt(4000, 7800),
  ratingCount: getRandomInt(5, 250),
  description: getRandomArrayItems([`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`, `Gustave prides himself on providing first-class service to the hotel&apos;s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave&aposs lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`, `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`, `Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante. Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum. Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`, `Sed sed nisi sed augue convallis suscipit in sed felis. Aliquam erat volutpat. Nunc fermentum tortor ac porta dapibus. In rutrum ac purus sit amet tempus.`], getRandomInt(2, 5)),
  director: getRandomArrayItems([
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`
  ], 1),
  starring: getRandomArrayItems([
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`
  ], getRandomInt(3, 12)),
  reviews: getMockReviews(getRandomInt(3, 7))
});

const getMockReviews = (count) => new Array(count).fill(``).map(() => getReviews());

const getReviews = () => ({
  author: getRandomArrayItems([
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`
  ], 1),
  rating: `${getRandomInt(0, 9)}.${getRandomInt(0, 9)}`,
  date: getRandomInt(1262293200000, 1552915226386),
  comment: getRandomArrayItems([
    `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
  ], 1)
});

const promoFilmMock = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`
};

const mockFilmForTests = {
  id: 0,
  title: `Revenant`,
  img: `img/revenant.jpg`,
  genre: `Dramas`,
  rating: `8.5`,
  src: `https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4`,
  releaseDate: 1552915226386,
  runTime: 5084,
  ratingCount: 148,
  description: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`, `Gustave prides himself on providing first-class service to the hotel&apos;s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave&aposs lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  director: `Bill Murray`,
  starring: [`Edward Norton`, `Jude Law`, `Willem Dafoe and other`, `Wes Andreson`],
  reviews: [
    {
      author: `Edward Norton`,
      rating: `3.5`,
      date: 1500713355882,
      comment: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
    },
    {
      author: `Willem Dafoe`,
      rating: `6.7`,
      date: 1302290042303,
      comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`
    },
    {
      author: `Ralph Fiennes`,
      rating: `8.9`,
      date: 1445604462862,
      comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`
    }
  ]
};

const mockFilmsForTests = [
  {
    id: 0,
    title: `Aviator`,
    img: `img/aviator.jpg`,
    genre: `Comedies`,
    rating: `8.1`,
    src: `https://archive.org/download/ElephantsDream/ed_1024_512kb.mp4`,
    releaseDate: 1552905226386,
    runTime: 6104,
    ratingCount: 198,
    description: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`],
    director: `Bill Murray`,
    starring: [`Edward Norton`, `Jude Law`, `Willem Dafoe and other`, `Wes Andreson`],
    reviews: [
      {
        author: `Edward Norton`,
        rating: `3.5`,
        date: 1500793355882,
        comment: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
      },
      {
        author: `Willem Dafoe`,
        rating: `6.7`,
        date: 1302908542303,
        comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`
      },
      {
        author: `Ralph Fiennes`,
        rating: `8.9`,
        date: 1444604462862,
        comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`
      }
    ]
  },
  {
    id: 1,
    title: `Macbeth`,
    img: `img/macbeth.jpg`,
    genre: `Horror`,
    rating: `3.6`,
    src: `https://interactive-examples.mdn.mozilla.net/media/examples/flower.mp4`,
    releaseDate: 1552005226386,
    ratingCount: 248,
    description: [`Gustave prides himself on providing first-class service to the hotel&apos;s guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave&aposs lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Edward Norton`,
    starring: [`Jude Law`, `Willem Dafoe and other`, `Wes Andreson`],
    reviews: [
      {
        author: `Edward Norton`,
        rating: `3.5`,
        date: 1500793355882,
        comment: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
      },
      {
        author: `Willem Dafoe`,
        rating: `6.7`,
        date: 1302908542303,
        comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`
      },
      {
        author: `Ralph Fiennes`,
        rating: `8.9`,
        date: 1444604462862,
        comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`
      }
    ]
  },
  {
    id: 2,
    title: `Snatch`,
    img: `img/snatch.jpg`,
    genre: `Documentary`,
    rating: `2.5`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    releaseDate: 1552518826386,
    runTime: 7100,
    ratingCount: 850,
    description: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.`, `Gustave prides himself on providing first-class, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
    director: `Jude Law`,
    starring: [`Edward Norton`, `Willem Dafoe and other`, `Wes Andreson`],
    reviews: [
      {
        author: `Edward Norton`,
        rating: `3.5`,
        date: 1500793355882,
        comment: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`
      },
      {
        author: `Willem Dafoe`,
        rating: `6.7`,
        date: 1302908542303,
        comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`
      },
      {
        author: `Ralph Fiennes`,
        rating: `8.9`,
        date: 1444604462862,
        comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`
      }
    ]
  }
];

const films = getFilms(FILM_COUNT);

export default films;
export {promoFilmMock, mockFilmForTests, mockFilmsForTests};
