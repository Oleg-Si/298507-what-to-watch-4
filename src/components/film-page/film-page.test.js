import React from 'react';
import renderer from 'react-test-renderer';
import {FilmPage} from './film-page.jsx';
import {mockFilmForTests} from '../../mock/films.js';
import {mockFilmsForTests} from '../../mock/films.js';
import {AuthorizationStatus} from './../../constants';

it(`Проверяет снепшот компонента FilmPage`, () => {
  const tree = renderer.create(
      <FilmPage
        films={mockFilmsForTests}
        activeFilm={mockFilmForTests}
        activeTab={`Overview`}
        onTabClick={() => {}}
        onFilmCardTitleClick={() => {}}
        onSignInClick={() => {}}
        onAddReview={() => {}}
        authorizationStatus={AuthorizationStatus.AUTH}
        userAvatar={`img/avatar.jpg`}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
