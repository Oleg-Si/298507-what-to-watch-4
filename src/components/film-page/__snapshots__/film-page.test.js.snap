// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверяет снепшот компонента FilmPage 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "rgb(146, 145, 139)",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Revenant"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Revenant.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            href="/mylist"
            onClick={[Function]}
          >
            <div
              className="user-block__avatar"
            >
              <img
                alt="User avatar"
                height="63"
                src="img/avatar.jpg"
                width="63"
              />
            </div>
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Revenant
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
                <symbol
                  id="play-s"
                  viewBox="0 0 19 19"
                >
                  <path
                    clipRule="evenodd"
                    d="M0 0L19 9.5L0 19V0Z"
                    fill="#EEE5B5"
                    fillRule="evenodd"
                  />
                </symbol>
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/films/0/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Revenant"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Revenant.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              8.5
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                148
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave&aposs friend and protege.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Bill Murray
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Edward Norton, Jude Law, Willem Dafoe and other, Wes Andreson
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2020 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
